index.d.ts

```diff
@@ -5779,8 +5779,8 @@ declare module sequelize {
      * @param options Transaction Options
      * @param autoCallback Callback for the transaction
      */
-    transaction(options: TransactionOptions, autoCallback: (t: Transaction) => PromiseLike<any>): SequelizePromise<any>;
-    transaction(autoCallback: (t: Transaction) => PromiseLike<any>): SequelizePromise<any>;
+    transaction<TResult>(options: TransactionOptions, autoCallback: (t: Transaction) => PromiseLike<TResult>): SequelizePromise<TResult>;
+    transaction<TResult>(autoCallback: (t: Transaction) => PromiseLike<TResult>): SequelizePromise<TResult>;
```

Comment:
Interesting, is this Sequelize runtime behaviour?

Reply:
@felixfbecker Yup! That's actually exactly how we've been been using it already, just with the current `<any>` type.